bin/install_swiftpm_dependencies

```diff
@@ -1,9 +1,65 @@
 #!/bin/bash -eu
 
+CUSTOM_PACKAGE_RESOLVED_PATH=${1-}
+BUILD_TYPE=${2-}
+SPM_CACHE_LOCATION="$HOME/spm"
+
+WORKSPACE_PACKAGE_RESOLVED_PATH=(*.xcworkspace/xcshareddata/swiftpm/Package.resolved)
+ROOT_PACKAGE_RESOLVED_PATH=Package.resolved
+
+# Find where Package.resolved is located
+if [[ -n "$CUSTOM_PACKAGE_RESOLVED_PATH" ]]; then
+	PACKAGE_RESOLVED_LOCATION=$CUSTOM_PACKAGE_RESOLVED_PATH
+elif  [[ -f ${WORKSPACE_PACKAGE_RESOLVED_PATH[0]} ]]; then
+	BUILD_TYPE="XCODEBUILD"
+	PACKAGE_RESOLVED_LOCATION=${WORKSPACE_PACKAGE_RESOLVED_PATH[0]}
+elif  [[ -f $ROOT_PACKAGE_RESOLVED_PATH ]]; then
+	BUILD_TYPE="SWIFT"
+	PACKAGE_RESOLVED_LOCATION=$ROOT_PACKAGE_RESOLVED_PATH
+else
+	echo "Unable to find Package.resolved"
+fi
+
+PACKAGE_RESOLVED_HASH=$(hash_file "$PACKAGE_RESOLVED_LOCATION")
+CACHE_KEY="$BUILDKITE_PIPELINE_SLUG-spm-cache-$PACKAGE_RESOLVED_HASH"
+
+# Restore SPM cache if it's available
+mkdir -p "$SPM_CACHE_LOCATION"
+cd "$SPM_CACHE_LOCATION"
+restore_cache "$CACHE_KEY"
+cd -
+
+# This will let Xcode use the system SSH config for downloading packages
 sudo defaults write com.apple.dt.Xcode IDEPackageSupportUseBuiltinSCM YES
 
 # Trust all GitHub.com and BitBucket.org keys – this allows checking out dependencies via SSH
 add_host_to_ssh_known_hosts bitbucket.org
 add_host_to_ssh_known_hosts github.com
 
-xcodebuild -resolvePackageDependencies
+# Based on the project type, resolve the packages using the correct build type
+if [[ "$BUILD_TYPE" == "XCODEBUILD" ]]; then
+	echo "Resolving packages with \`xcodebuild\`"
+	xcodebuild \
+	-resolvePackageDependencies \
+	-clonedSourcePackagesDirPath "$SPM_CACHE_LOCATION"
+elif [[ "$BUILD_TYPE" == "SWIFT" ]]; then
+	echo "Resolving packages with \`swift\`"
+	swift package resolve \
+	--cache-path "$SPM_CACHE_LOCATION" \
+	--verbose
+else 
+	echo "BUILD_TYPE is not set"
+fi
+
+# `checkouts` can be excluded because the system can quickly generate them 
+# instead of needing to download them in the cache each time.
+# 
+# `artifacts` should be excluded because it causes issues when downloading 
+# certain packages to have the artifacts already present after extracting 
+# cache
+TAR_OPTIONS="--exclude=./checkouts --exclude=./artifacts"
```

Comment:
Rather than fighting with bash syntax regarding using quote or not, I'm wondering if it'd be simpler to just delete these unwanted files from the SPM_CACHE_LOCATION?

Reply:
Or, copy the "SPM_CACHE_LOCATION", delete the unwanted files from the new copy, call `save_cache` on the new copy, and delete the new copy afterwards.

Reply:
@AliSoftware Thank you for the explanation! The array solution makes a lot of sense after the behavior I saw while debugging this.

@crazytonyli Good idea! I think I'm going to go with deleting those directories before saving the cache. It removes the complexity of passing the tar options along with simplifying the changes to `save_cache`.